pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -205,10 +209,19 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 	errors := map[*corev1.Pod]error{}
 	QueueDepth.DeletePartialMatch(prometheus.Labels{ControllerLabel: injection.GetControllerName(ctx)}) // Reset the metric for the controller, so we don't keep old ids around
 	q := NewQueue(pods...)
+
+	startTime := s.clock.Now()
+	lastLogTime := s.clock.Now()
+	batchSize := len(q.pods)
 	for {
 		QueueDepth.With(
 			prometheus.Labels{ControllerLabel: injection.GetControllerName(ctx), schedulingIDLabel: string(s.id)},
 		).Set(float64(len(q.pods)))
+
+		if s.clock.Since(lastLogTime) > time.Minute {
+			log.FromContext(ctx).Info("printing scheduling progress", "pods-scheduled", batchSize-len(q.pods), "pods-left", len(q.pods), "time-spent", s.clock.Since(startTime).Truncate(time.Second))
```

Comment:
```suggestion
			log.FromContext(ctx).Info("printing scheduling progress", "pods-scheduled", batchSize-len(q.pods), "pods-left", len(q.pods), "duration", s.clock.Since(startTime).Truncate(time.Second))
```

I still think it's more ergonomic if we go with "scheduling pods..." to indicate that we are still making progress. Also `duration` feels more ergonomic to me compared to `time-spent` since it's consistent with wording elsewhere in the code.

Also, can we see an example of this happening? I'm also curious if we want to just show this during provisioning -- I'm worried that this is going to be extra noisy with provisioning and disruption running at the same time.

Reply:
This fires once a minute, per provisioning controller. Don't think it'll be too noisy.

Reply:
Per scheduler -- not provisioning controller and per-scheduling loop. With multiple scheduling loops, I think this could get extremely noisy